lib/WorseReflection/Bridge/Phpactor/DocblockParser/PHPStanTypeConverter.php

```diff
@@ -0,0 +1,259 @@
+use Phpactor\WorseReflection\Core\Type\CallableType;
+use Phpactor\WorseReflection\Core\Type\ClassStringType;
+use Phpactor\WorseReflection\Core\Type\ClassType;
+use Phpactor\WorseReflection\Core\Type\ClosureType;
+use Phpactor\WorseReflection\Core\Type\ConditionalType;
+use Phpactor\WorseReflection\Core\Type\FalseType;
+use Phpactor\WorseReflection\Core\Type\FloatLiteralType;
+use Phpactor\WorseReflection\Core\Type\FloatType;
+use Phpactor\WorseReflection\Core\Type\GenericClassType;
+use Phpactor\WorseReflection\Core\Type\GlobbedConstantUnionType;
+use Phpactor\WorseReflection\Core\Type\IntLiteralType;
+use Phpactor\WorseReflection\Core\Type\IntMaxType;
+use Phpactor\WorseReflection\Core\Type\IntNegative;
+use Phpactor\WorseReflection\Core\Type\IntPositive;
+use Phpactor\WorseReflection\Core\Type\IntRangeType;
+use Phpactor\WorseReflection\Core\Type\IntType;
+use Phpactor\WorseReflection\Core\Type\IntersectionType;
+use Phpactor\WorseReflection\Core\Type\InvokeableType;
+use Phpactor\WorseReflection\Core\Type\ListType;
+use Phpactor\WorseReflection\Core\Type\MissingType;
+use Phpactor\WorseReflection\Core\Type\MixedType;
+use Phpactor\WorseReflection\Core\Type\NeverType;
+use Phpactor\WorseReflection\Core\Type\NullType;
+use Phpactor\WorseReflection\Core\Type\NullableType;
+use Phpactor\WorseReflection\Core\Type\ObjectType;
+use Phpactor\WorseReflection\Core\Type\PseudoIterableType;
+use Phpactor\WorseReflection\Core\Type\ReflectedClassType;
+use Phpactor\WorseReflection\Core\Type\ResourceType;
+use Phpactor\WorseReflection\Core\Type\SelfType;
+use Phpactor\WorseReflection\Core\Type\StaticType;
+use Phpactor\WorseReflection\Core\Type\StringLiteralType;
+use Phpactor\WorseReflection\Core\Type\StringType;
+use Phpactor\WorseReflection\Core\Type\ThisType;
+use Phpactor\WorseReflection\Core\Type\UnionType;
+use Phpactor\WorseReflection\Core\Type\VoidType;
+use Phpactor\WorseReflection\Reflector;
+
+class PHPStanTypeConverter
+{
+    public function __construct(private Reflector $reflector, private ?ReflectionScope $scope)
+    {
+    }
+
+    public function convert(?TypeNode $type): Type
+    {
+        return match (true) {
+            $type instanceof ArrayShapeNode => $this->convertArrayShape($type),
+            $type instanceof ArrayTypeNode => new ArrayType($this->convert($type->type)),
+            $type instanceof CallableTypeNode => $this->convertCallable($type),
+            $type instanceof ConditionalTypeForParameterNode => $this->convertConditional($type),
+            $type instanceof ConstTypeNode => $this->convertConstant($type),
+            $type instanceof GenericTypeNode => $this->convertGeneric($type),
+            $type instanceof IdentifierTypeNode => $this->convertSimpleIdentifier($type->name),
+            $type instanceof IntersectionTypeNode => $this->convertIntersection($type),
+            $type instanceof NullableTypeNode => new NullableType($this->convert($type->type)),
+            // todo: ObjectShape
```

Comment:
I'm not sure if it's ok to create this along this PR, but IMHO it would be easy to implement some additional types, which are missing now, like object shape or `non-empty-string`, etc. following https://phpstan.org/writing-php-code/phpdoc-types

Reply:
yes, but I would aim for the minimum required for green tests before adding more stuff.